x-pack/plugins/security_solution/server/lib/detection_engine/rule_execution_log/event_log/event_log_reader.ts

```diff
@@ -31,6 +55,64 @@ export interface GetLastStatusChangesArgs {
 
 export const createEventLogReader = (eventLog: IEventLogClient): IEventLogReader => {
   return {
+    async getAggregateExecutionEvents(
+      args: GetAggregateExecutionEventsArgs
+    ): Promise<GetAggregateRuleExecutionEventsResponse> {
+      const { ruleId, start, end, statusFilters, page, perPage, sortField, sortOrder } = args;
+      const soType = RULE_SAVED_OBJECT_TYPE;
+      const soIds = [ruleId];
+
+      // Current workaround to support root level filters without missing fields in the aggregate event
+      // or including events from statuses that aren't selected
+      // TODO: See: https://github.com/elastic/kibana/pull/127339/files#r825240516
+      // First fetch execution uuid's by status filter if provided
+      let statusIds: string[] = [];
+      // If 0 or 3 statuses are selected we can search for all statuses and don't need this pre-filter by ID
+      if (statusFilters.length > 0 && statusFilters.length < 3) {
+        // TODO: Add cardinality agg and pass as maxEvents in response
+        const statusResults = await eventLog.aggregateEventsBySavedObjectIds(soType, soIds, {
+          start,
+          end,
+          filter: `kibana.alert.rule.execution.status:(${statusFilters.join(' OR ')})`,
+          aggs: {
+            filteredExecutionUUIDs: {
+              terms: {
+                field: 'kibana.alert.rule.execution.uuid',
+                size: MAX_EXECUTION_EVENTS_DISPLAYED,
+              },
+            },
+          },
+        });
+        const filteredExecutionUUIDs = statusResults.aggregations
+          ?.filteredExecutionUUIDs as ExecutionUuidAggResult;
+        statusIds = filteredExecutionUUIDs?.buckets?.map((b) => b.key) ?? [];
+        // Early return if no results based on status filter
+        if (statusIds.length === 0) {
+          return {
+            total: 0,
+            events: [],
+          };
+        }
+      }
+
+      // Now query for aggregate events, and pass any ID's as filters as determined from the above status/queryText results
+      const idsFilter = statusIds.length
+        ? `kibana.alert.rule.execution.uuid:(${statusIds.join(' OR ')})`
+        : '';
+      const results = await eventLog.aggregateEventsBySavedObjectIds(soType, soIds, {
+        start,
+        end,
+        filter: idsFilter,
+        aggs: getExecutionEventAggregation({
+          maxExecutions: MAX_EXECUTION_EVENTS_DISPLAYED,
+          page,
+          perPage,
+          sort: [{ [sortField]: { order: sortOrder } }] as estypes.Sort,
```

Comment:
Hopefully remove as part of decode on route

Reply:
Taken care of in d205841, decoded types from route now all match their use so no need for casting 🙂